Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -79,19 +79,19 @@ void PPCSymbolDB::AddKnownSymbol(u32 startAddr, u32 size, const std::string& nam
 
 Symbol* PPCSymbolDB::GetSymbolFromAddr(u32 addr)
 {
-  XFuncMap::iterator it = functions.find(addr);
-  if (it != functions.end())
-  {
+  XFuncMap::iterator it = functions.lower_bound(addr);
+  if (it == functions.end())
+    return nullptr;
+
+  // If the address is exactly the start address of a symbol, we're done.
+  if (it->second.address == addr)
     return &it->second;
-  }
-  else
-  {
-    for (auto& p : functions)
-    {
-      if (addr >= p.second.address && addr < p.second.address + p.second.size)
-        return &p.second;
-    }
-  }
+
+  // Otherwise, check whether the address is within the bounds of a symbol.
+  --it;
```

Comment:
if (it != functions.end())

Reply:
it can't be functions.end(), because we return nullptr from the function if it points to the end.